pkg/embeddedcluster/roles.go

```diff
@@ -112,6 +126,60 @@ func labelify(s string) string {
 	return trimmed
 }
 
+// borrowed from https://github.com/kubernetes/apimachinery/blob/9254095ca5cab3666d500ec67cd00f9ab0d113d7/pkg/util/validation/validation.go#L206-L208
+
+const dns1123LabelFmt string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
+const dns1123SubdomainFmt string = dns1123LabelFmt + "(\\." + dns1123LabelFmt + ")*"
+
+// DNS1123SubdomainMaxLength is a subdomain's max length in DNS (RFC 1123)
+const dns1123SubdomainMaxLength int = 253
+
+var dns1123SubdomainRegexp = regexp.MustCompile("^" + dns1123SubdomainFmt + "$")
+
+// IsValidRFC1123 tests for a string that conforms to the definition of a
+// subdomain in DNS (RFC 1123).
+func IsValidRFC1123(value string) bool {
+	if len(value) > dns1123SubdomainMaxLength {
+		return false
+	}
+	if !dns1123SubdomainRegexp.MatchString(value) {
+		return false
+	}
+	return true
+}
+
+var dns1123IllegalStartRegex = regexp.MustCompile(`^[^0-9a-z]+`)
+var dns1123IllegalEndRegex = regexp.MustCompile(`[^0-9a-z]+$`)
+var dns1123IllegalCharsRegex = regexp.MustCompile(`[^0-9a-z-]`)
+
+func ConvertToRFC1123(value string, args ...int) string {
+	value = strings.ToLower(value)
+
+	if len(value) == 0 || IsValidRFC1123(value) {
+		return strings.ToLower(value)
```

Comment:
I think we can simply return `value` here.

Reply:
looks like it since we've already run `value = strings.ToLower(value)` at that point